selftests/unit/test_remote.py

```diff
@@ -25,154 +30,71 @@ class RemoteTestRunnerTest(unittest.TestCase):
-                        env_keep=None)
-        log = flexmock()
-        log.should_receive("info")
-        job = flexmock(args=Args, log=log)
-        self.runner = avocado_runner_remote.RemoteTestRunner(job, None)
-
-    def tearDown(self):
-        flexmock_teardown()
-
-    def test_setup(self):
-        """ Tests RemoteResult.test_setup() """
-        self.runner.setup()
-        flexmock_teardown()
+        """
+        Test RemoteTestRunner.run_suite()
+
+        The general idea of this test is to:
+
+        1) Create the machinery necessary to get a RemoteTestRunner
+           setup inside a job, or looking at it the other way around, to
+           have a runner that is created with a valid job.
+
+        2) Mock the interactions with a remote host.  This is done here
+           basically by mocking 'Remote' and 'fabric' usage.
+
+        3) Provide a polluted JSON to be parsed by the RemoteTestRunner
+
+        4) Assert that those results are properly parsed into the
+           job's result
+        """
+        job_args = argparse.Namespace(test_result_total=1,
+                                      remote_username='username',
+                                      remote_hostname='hostname',
+                                      remote_port=22,
+                                      remote_password='password',
+                                      remote_key_file=None,
+                                      remote_timeout=60,
+                                      show_job_log=False,
+                                      mux_yaml=['~/avocado/tests/foo.yaml',
+                                                '~/avocado/tests/bar/baz.yaml'],
+                                      dry_run=True,
+                                      env_keep=None,
+                                      reference=['/tests/sleeptest.py',
+                                                 '/tests/other/test',
+                                                 'passtest.py'])
+
+        job = Job(job_args)
+        runner = avocado_runner_remote.RemoteTestRunner(job, job.result)
+        runner.check_remote_avocado = mock.Mock(return_value=(True,
+                                                              (version.MAJOR,
+                                                               version.MINOR)))
+
+        # These are mocked at their source, and will prevent fabric from
+        # trying to contact remote hosts
+        with mock.patch('avocado_runner_remote.Remote'):
+            with mock.patch('avocado_runner_remote.fabric'):
+                runner.remote = avocado_runner_remote.Remote(job_args.remote_hostname)
+
+            # This is the result that the run_suite() will get from remote.run
+            remote_run_result = process.CmdResult()
+            remote_run_result.stdout = JSON_RESULTS
+            remote_run_result.exit_status = 0
+            runner.remote.run = mock.Mock(return_value=remote_run_result)
```

Comment:
Would it be possible to check the arguments which which the `run` is called? It could unveil some issues when we break the command.

Reply:
Yes, we can add `runner.remote.run.assert_called_with()`.

Reply:
Adding this on v2:

```diff
+            cmd_line = ('avocado run --force-job-id '
+                        '0000000000000000000000000000000000000000 --json - '
+                        '--archive /tests/sleeptest.py /tests/other/test '
+                        'passtest.py -m ~/avocado/tests/foo.yaml '
+                        '~/avocado/tests/bar/baz.yaml --dry-run')
+            runner.remote.run.assert_called_with(cmd_line,
+                                                 ignore_status=True,
+                                                 timeout=61)
```